cc-test/build.rs

```diff
@@ -103,4 +103,12 @@ fn main() {
     let out = cc::Build::new().file("src/expand.c").expand();
     let out = String::from_utf8(out).unwrap();
     assert!(out.contains("hello world"));
+
+    // Compilations _should_ fail, but we don't want it to show up as a warning.
+    cc::Build::new()
+        .cargo_metadata(false)
+        .cargo_warnings(false)
+        .file("src/suppress_warnings.c")
+        .try_compile("suppress_warnings")
+        .unwrap_err();
 }
```

Comment:
At the moment this "test" can't really be enforced. If it fails, you will see a warning in the output, but AFAIK there's no way to force that into an error.
Do you have an idea how else we could design a test for this?

Reply:
I think you can "fork" the process by writing a test that execute `std::env::current_exe()` then capture stderr/stdout